pkg/controller/job/job_controller.go

```diff
@@ -865,9 +867,9 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (forget bool, rEr
 	// improve the Job backoff policy when parallelism > 1 and few Jobs failed but others succeed.
 	// In this case, we should clear the backoff delay.
 	forget = job.Status.Succeeded < succeeded
+	needsStatusUpdate := suspendCondChanged || active != job.Status.Active || !equalReady(ready, job.Status.Ready) || finishedCondition != nil
```

Comment:
I would suggest to only update the new path `uncounted != nil` (as also described in the Issue / PR) unless for some good reasons.